dockerfiles/Dockerfile.server

```diff
@@ -0,0 +1,34 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#--------------------------------------------------------------------------
+# Official user quickstart docker container for ONNX Runtime CPU
+# Ubuntu 16.04, CPU version, Python 3.
+#--------------------------------------------------------------------------
+
+FROM ubuntu:16.04 AS minimal
+MAINTAINER Harry Yang "huayang@microsoft.com"
+
+FROM ubuntu:16.04 AS build
+ARG PYTHON_VERSION=3.5
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update && \
+    apt-get install -y sudo git bash
+
+ENV PATH="/opt/cmake/bin:${PATH}"
+RUN git clone --recursive https://github.com/Microsoft/onnxruntime onnxruntime
```

Comment:
better to support build from a specific branch/tag using --branch option with a branch ARG defaulting to master
you can also use --single-branch to avoid pulling more than what's needed and delete the source/build dirs after copying the server executable.

Reply:
@jywu-msft yeah, i had this comment earlier and then deleted it when i saw that the existing dockerfiles don't support branches as well.

Reply:
we're in the process of revamping the Dockerfiles. and they will all support branch ARGs

Reply:
Specify the branch using an ARG default to master and single-branch. Fixed